src/controllers/trainee/Controller.ts

```diff
@@ -0,0 +1,106 @@
+import { Request, Response } from 'express';
+import UserRepository from '../../repositories/user/UserRepository';
+import SystemResponse from '../../libs/routes/SystemResponse';
+import { IRequest } from '../../libs/interface';
+import { initSearch } from '../../libs/utils/helper';
+
+class TraineeController {
+  static instance: TraineeController;
+  public userRepo = new UserRepository();
+  private systemResponse: SystemResponse = new SystemResponse();
+  static getInstance(): TraineeController {
+    if (TraineeController.instance instanceof TraineeController) {
+      return TraineeController.instance;
+    }
+    return (TraineeController.instance = new TraineeController());
+  }
+  private constructor() { }
+
+  create = async (req: IRequest, res: Response): Promise<Response> => {
+    try {
+      const userId = req.user.userId;
+      const data = req.body;
+      const record = { ...data, userId };
+      const result = await this.userRepo.create(record);
+      if (result) {
+        delete result._id;
+        return this.systemResponse.success(
+          res,
+          `Trainee added Successfully`,
+          200,
+          result
+        );
+      }
+      return this.systemResponse.success(res, `Unauthorized User`, 403, result);
```

Comment:
How you can say the user is Unauthorized?

Reply:
Done